romancal/resample/resample.py

```diff
@@ -759,3 +763,73 @@ def gwcs_into_l3(model, wcsinfo):
     # l3_wcsinfo.dec_corn4 = from bounding box
     # l3_wcsinfo.orientat = ???
     # l3_wcsinfo.orientat_local = ???
+
+
+def populate_mosaic_basic(
+    output_model: datamodels.MosaicModel, input_models: [List, ModelContainer]
+):
+    """
+    Populate basic metadata fields in the output mosaic model based on input models.
+
+    Parameters
+    ----------
+    output_model : MosaicModel
+        Object to populate with basic metadata.
+    input_models : [List, ModelContainer]
+        List of input data models from which to extract the metadata.
+        ModelContainer is also supported.
+
+    Returns
+    -------
+    None
+    """
+
+    input_meta = [datamodel.meta for datamodel in input_models]
+
+    # time data
+    output_model.meta.basic.time_first_mjd = np.min(
+        [x.exposure.start_time.mjd for x in input_meta]
+    )
+    output_model.meta.basic.time_last_mjd = np.max(
+        [x.exposure.end_time.mjd for x in input_meta]
+    )
+    output_model.meta.basic.time_mean_mjd = np.mean(
+        [x.exposure.mid_time.mjd for x in input_meta]
+    )
+
+    # observation data
+    output_model.meta.basic.visit = (
+        input_meta[0].observation.visit
+        if len({x.observation.visit for x in input_meta}) == 1
+        else -1
+    )
+    output_model.meta.basic.segment = (
+        input_meta[0].observation.segment
+        if len({x.observation.segment for x in input_meta}) == 1
+        else -1
+    )
+    output_model.meta.basic["pass"] = (
+        input_meta[0].observation["pass"]
+        if len({x.observation["pass"] for x in input_meta}) == 1
+        else -1
+    )
+    output_model.meta.basic.program = (
+        input_meta[0].observation.program
+        if len({x.observation.program for x in input_meta}) == 1
+        else "-1"
+    )
+    output_model.meta.basic.survey = (
+        input_meta[0].observation.survey
+        if len({x.observation.survey for x in input_meta}) == 1
+        else "MULTIPLE"
+    )
+
+    # instrument data
+    output_model.meta.basic.optical_element = input_meta[0].instrument.optical_element
+    output_model.meta.basic.instrument = input_meta[0].instrument.name
+
+    # skycell location
+    output_model.meta.basic.location_name = "TBD"
+
```

Comment:
Why is this location_name? Why not skycell_name? 
will we need the name of the projection call as well?

Reply:
That's the attribute's name in RAD; I concur that `skycell_name` sounds better than `location_name`.

Reply:
RTB hadn't (and still hasn't, as far as I know) decided if we want to call these skycells or patches or ....  We ended up with "location_name" as some kind of generic word.  I'd propose we fill this with the record name from the tessellation file, which includes both the projection cell part and the sky cell part.

We'll specify product_type = 'PASS' to handle passes, etc., and the pass / visit metadata will get sentinel values indicating that they span multiple visits, as in Mairan's current PR.